packaging/repositories/_utils.py

```diff
@@ -5,44 +5,81 @@
 
 import functools
 
-import attr
+from .base import Request, Response
+from effect import Effect, sync_perform, sync_performer, ParallelEffects
+from effect.async import perform_parallel_async as perform_parallel_serially
+from effect import base_dispatcher, ComposedDispatcher, TypeDispatcher
 
-from .base import Response
 
+from txeffect import (
+    make_twisted_dispatcher, deferred_performer, perform as twisted_performer)
+from twisted.internet.defer import inlineCallbacks, returnValue
 
-@attr.s(cmp=False)
-class _RepositoryFetcher:
 
-    _coro = attr.ib()
-    _finished = attr.ib(default=False, init=False)
-    _result = attr.ib(default=None, init=False)
+@sync_performer
+def _blocking_request(dispatcher, intent):
+    from requests import get
+    response = get(intent.url)
+    return Response(
+        content=response.content,
+        headers=response.headers,
+    )
 
-    @property
-    def finished(self):
-        return self._finished
+_blocking_dispatcher = ComposedDispatcher([
+    TypeDispatcher({Request: _blocking_request
+                    ParallelEffects, perform_parallel_serially,
```

Comment:
This could instead be https://effect.readthedocs.io/en/latest/api/effect.threads.html to use threads to run requests in parallel.